src/test/scala/org/broadinstitute/dsde/rawls/integrationtest/HttpGooglePubSubDAOSpec.scala

```diff
@@ -0,0 +1,105 @@
+import org.scalatest.{BeforeAndAfterAll, FlatSpec, Matchers}
+
+import scala.concurrent.duration.Duration
+import scala.concurrent.{Await, Future}
+
+class HttpGooglePubSubDAOSpec extends FlatSpec with Matchers with IntegrationTestConfig with BeforeAndAfterAll with Retry with TestDriverComponent {
+  implicit val system = ActorSystem("HttpGooglePubSubDAOSpec")
+  val gpsDAO = new HttpGooglePubSubDAO(
+    GoogleClientSecrets.load(
+      JacksonFactory.getDefaultInstance, new StringReader(gcsConfig.getString("secrets"))),
+    gcsConfig.getString("pathToPem"),
+    gcsConfig.getString("appName"),
+    gcsConfig.getString("serviceProject")
+  )
+
+  def cleanup() = {
+    Await.ready(gpsDAO.deleteSubscription("test-subscription"), Duration.Inf)
+    Await.ready(gpsDAO.deleteTopic("integration-tests"), Duration.Inf)
+  }
+
+  "HttpGooglePubSubDAOSpec" should "do all of the things" in {
+
+    //if this test ever fails, there will likely be some leftover garbage. clean it up first
+    cleanup()
+
+    //create a topic and subscription
+    Await.result(gpsDAO.createTopic("integration-tests"), Duration.Inf)
+    Await.result(gpsDAO.createSubscription("integration-tests", "test-subscription"), Duration.Inf)
+
+    //publish a few messages to the topic
+    val messages = Seq("test-1", "test-2", "test-3", "test-4", "test-5")
+    Await.result(gpsDAO.publishMessages("integration-tests", messages), Duration.Inf)
+
+    Await.result(gpsDAO.withMessage("test-subscription") { msg =>
+      assertResult(true) {
+        messages.contains(msg)
+      }
+      Future.successful(gpsDAO.MessageAcknowledged)
+    }, Duration.Inf)
+
+    //cleanup
+    Await.result(gpsDAO.deleteSubscription("test-subscription"), Duration.Inf)
+    Await.result(gpsDAO.deleteTopic("integration-tests"), Duration.Inf)
+  }
+
+  it should "gracefully handle there being no messages in the queue" in {
+
+    //if this test ever fails, there will likely be some leftover garbage. clean it up first
+    cleanup()
+
+    //create a topic and subscription
+    Await.result(gpsDAO.createTopic("integration-tests"), Duration.Inf)
+    Await.result(gpsDAO.createSubscription("integration-tests", "test-subscription"), Duration.Inf)
+
+    Await.result(gpsDAO.withMessage("test-subscription") { msg =>
+      assertResult(None) {
+        msg
+      }
+      Future.successful(gpsDAO.MessageNotAcknowledged)
+    }, Duration.Inf)
+
+    //cleanup
+    Await.result(gpsDAO.deleteSubscription("test-subscription"), Duration.Inf)
+    Await.result(gpsDAO.deleteTopic("integration-tests"), Duration.Inf)
+  }
+
+  it should "do all of the things with multiple messages" in {
+
+    //if this test ever fails, there will likely be some leftover garbage. clean it up first
+    cleanup()
+
+    //create a topic and subscription
+    Await.result(gpsDAO.createTopic("integration-tests"), Duration.Inf)
+    Await.result(gpsDAO.createSubscription("integration-tests", "test-subscription"), Duration.Inf)
```

Comment:
These first five lines (code and comments) and the last three lines are duplicated in each of these tests. Can you find a way to not repeat yourself? Something like a `with-test-topics-and-subscriptions` maybe?